merkletree/Cargo.toml

```diff
@@ -5,9 +5,14 @@ version.workspace = true
 edition.workspace = true
 rust-version.workspace = true
 
+[features]
+scale-codec-encode = ["dep:parity-scale-codec"]
+scale-codec-decode = ["dep:parity-scale-codec"]
```

Comment:
It's maybe a bit too granular to have this as two separate features. Either one will pull in the dependency, and enabling both does not add any new deps. When crated optionally depend on `serde`, both encoding and decoding are guarded by single feature. Do you have a specific reason to do it like this?